src/apps/bridge/bridgePowerController.cpp

```
@@ -110,34 +113,73 @@ bool BridgePowerController::waitForSignal(bool on, TickType_t ticks_to_wait) {
   return rval;
 }
 
-void BridgePowerController::powerBusAndSetSignal(bool on, bool notifyL2) {
+/*!
+ * \brief Controls bridge bus power with proper sequencing and event signaling.
+ *
+ * Power-up sequence (when busOn=true):
+ * 1. Turn on boost converter first
+ * 2. Wait for capacitors to charge (CAPACITOR_CHARGE_DELAY_MS)
+ * 3. Turn on load switch to enable bus power
+ *
+ * Power-down sequence (when busOn=false):
+ * 1. Turn off load switch first to disconnect the load
+ * 2. Turn off boost converter (saves ~3mW)
+ *
+ * The function also:
+ * - Checks current pin state to avoid unnecessary operations
+ * - Sets/clears event group bits to signal power state changes to waiting tasks
+ * - Optionally notifies L2 network layer of power state changes
+ * - Disables unused port 2 during power-up for power savings
+ * - Logs power state changes for debugging
+ *
+ * \param[in] busOn - true to turn on bus power, false to turn off
+ * \param[in] notifyL2 - true to notify L2 network layer of power changes (default: true)
+ *
+ * \note The boost converter must be turned on before the load switch to allow
+ *       capacitor charging time. See PR#335 for detailed analysis.
+ * \note Port 2 is automatically disabled during power-up as it's unused on bridge
+ *       hardware and disabling provides power savings.
+ */
+void BridgePowerController::setBusPowerAndSetSignal(bool busOn, bool notifyL2) {
```

Should we return an error code here? 

I think several items can fail: IOWrite, IORead, xEventGroupClearBits